src/Service.Tests/Unittests/RuntimeConfigLoaderJsonDeserializerTests.cs

```diff
@@ -0,0 +1,267 @@
+    /// </summary>
+    [TestClass, TestCategory(TestCategory.MSSQL)]
+    public class RuntimeConfigLoaderJsonDeserializerTests
+    {
+        #region Positive Tests
+
+        /// <summary>
+        /// Test valid cases for parsing the runtime config.
+        /// These cases have strings close to the pattern we
+        /// match when looking to replace parts of the config,
+        /// strings that match said pattern, and other edge
+        /// cases to reveal if the pattern matching is working.
+        /// The pattern we look to match is @env('') where we take
+        /// what is inside of the '', ie: @env('<match>'). The match is then
+        /// used to get the associated environment variable.
+        /// </summary>
+        /// <param name="repKeys">Replacement used as key to get environment variable.</param>
+        /// <param name="repValues">Replacement value.</param>
+        [DataTestMethod]
+        [DataRow(
+            new string[] { "@env(')", "@env()", "@env(')'@env('()", "@env('@env()'", "@@eennvv((''''))" },
+            new string[] { "@env(')", "@env()", "@env(')'@env('()", "@env('@env()'", "@@eennvv((''''))" },
+            DisplayName = "Replacement strings that won't match.")]
+        [DataRow(
+            new string[] { "@env('envVarName')", "@env(@env('envVarName'))", "@en@env('envVarName')", "@env'()@env'@env('envVarName')')')" },
+            new string[] { "envVarValue", "@env(envVarValue)", "@enenvVarValue", "@env'()@env'envVarValue')')" },
+            DisplayName = "Replacement strings that match.")]
+        //  since we match strings surrounded by single quotes,
+        //  the following are environment variable names set to the
+        //  associated values:
+        // 'envVarName  -> _envVarName
+        //  envVarName' ->  envVarName_
+        // 'envVarName' -> _envVarName_
+        [DataRow(
+            new string[] { "@env(')", "@env()", "@env('envVarName')", "@env(''envVarName')", "@env('envVarName'')", "@env(''envVarName'')" },
+            new string[] { "@env(')", "@env()", "envVarValue", "_envVarValue", "envVarValue_", "_envVarValue_" },
+            DisplayName = "Replacement strings with some matches.")]
+        public void CheckConfigEnvParsingTest(string[] repKeys, string[] repValues)
+        {
+            SetEnvVariables();
+            Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(GetModifiedJsonString(repValues), out RuntimeConfig expectedConfig), "Should read the expected config");
+            Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(GetModifiedJsonString(repKeys), out RuntimeConfig actualConfig), "Should read actual config");
+
+            Assert.AreEqual(expectedConfig.ToJson(), actualConfig.ToJson());
+        }
+
+        /// <summary>
+        /// Method to validate that comments are skipped in config file (and are ignored during deserialization).
+        /// </summary>
+        [TestMethod]
+        public void CheckCommentParsingInConfigFile()
```

Comment:
Do we support comments in JSON config now? That's awesome!! There's an existing issue which can be closed if thats the case.

Reply:
Yep, they are supported but they won't be written back out as that's just a limitation of System.Text.Json (see https://github.com/dotnet/runtime/labels/area-System.Text.Json and search for comments).